core/templates/dev/head/pages/exploration_editor/editor_tab/TrainingDataEditorPanelService.js

```diff
@@ -146,7 +149,9 @@ oppia.factory('TrainingDataEditorPanelService', [
               };
 
               $scope.openTrainUnresolvedAnswerModal = function(answerIndex) {
-                if ($scope.trainingData.length > 1) {
+                if (($scope.answerGroupHasNonEmptyRules &&
+                    $scope.trainingData.length > 0) ||
```

Comment:
Not sure I understand this criterion. Could you explain (maybe in a comment)?

Reply:
```
// An answer group must have either a rule or at least one
// answer in training data. Don't allow modification of training
// data answers if there are no rules and only one training data
// answer is present.
```

Reply:
Ah ok, that's a good comment, thanks!